### ISSUE TITLE
toBeObject matches also array

### ISSUE THREAD

**Issue by cs4inctec** [jest-community/jest-extended#148]
**Bug**

- `package` version: 0.7.2
- `node` version: 8.11.2
- `npm` version: 5.6.0

Relevant code or config

```javascript
test('should not match', () => {
    expect([1, 2, 3, 4]).toBeObject();
})
```

What you did:
I tested if an element is an object and printed it to screen. 

What happened (please provide anything you think will help):
Output was an array, but the test did not fail.
I expected that an array would fail on toBeObject.

**Comment by benjaminkay93** [comment]
I ran this myself, it also passes on node 10.5 with npm 6.1, I don't think this is expected behaviour

**Comment by andrew-yustyk** [comment]
Seems like the quickest solution is to add `!Array.isArray(expected)` condition to the toBeObject matcher

**Comment by mattphillips** [comment]
Fixed in 230d07120a902362e64e27d5d2da28fb2956feb7
